pkg/kubectl/scheme/install.go

```diff
@@ -55,238 +55,24 @@ import (
-		},
-	).Register(Registry, Scheme); err != nil {
-		panic(err)
-	}
-
-	Versions = append(Versions, policyv1beta1.SchemeGroupVersion)
-	if err := announced.NewGroupMetaFactory(
-		&announced.GroupMetaFactoryArgs{
-			GroupName:              policyv1beta1.GroupName,
-			VersionPreferenceOrder: []string{policyv1beta1.SchemeGroupVersion.Version},
-		},
-		announced.VersionToSchemeFunc{
-			policyv1beta1.SchemeGroupVersion.Version: policyv1beta1.AddToScheme,
-		},
-	).Register(Registry, Scheme); err != nil {
-		panic(err)
-	}
-
-	Versions = append(Versions, rbacv1.SchemeGroupVersion, rbacv1beta1.SchemeGroupVersion, rbacv1alpha1.SchemeGroupVersion)
-	if err := announced.NewGroupMetaFactory(
-		&announced.GroupMetaFactoryArgs{
-			GroupName:              rbacv1.GroupName,
-			VersionPreferenceOrder: []string{rbacv1.SchemeGroupVersion.Version, rbacv1beta1.SchemeGroupVersion.Version, rbacv1alpha1.SchemeGroupVersion.Version},
-		},
-		announced.VersionToSchemeFunc{
-			rbacv1.SchemeGroupVersion.Version:       rbacv1.AddToScheme,
-			rbacv1beta1.SchemeGroupVersion.Version:  rbacv1beta1.AddToScheme,
-			rbacv1alpha1.SchemeGroupVersion.Version: rbacv1alpha1.AddToScheme,
-		},
-	).Register(Registry, Scheme); err != nil {
-		panic(err)
-	}
-
-	Versions = append(Versions, schedulingv1alpha1.SchemeGroupVersion)
-	if err := announced.NewGroupMetaFactory(
-		&announced.GroupMetaFactoryArgs{
-			GroupName:              schedulingv1alpha1.GroupName,
-			VersionPreferenceOrder: []string{schedulingv1alpha1.SchemeGroupVersion.Version},
-		},
-		announced.VersionToSchemeFunc{
-			schedulingv1alpha1.SchemeGroupVersion.Version: schedulingv1alpha1.AddToScheme,
-		},
-	).Register(Registry, Scheme); err != nil {
-		panic(err)
-	}
-
-	Versions = append(Versions, settingsv1alpha1.SchemeGroupVersion)
-	if err := announced.NewGroupMetaFactory(
-		&announced.GroupMetaFactoryArgs{
-			GroupName:              settingsv1alpha1.GroupName,
-			VersionPreferenceOrder: []string{settingsv1alpha1.SchemeGroupVersion.Version},
-		},
-		announced.VersionToSchemeFunc{
-			settingsv1alpha1.SchemeGroupVersion.Version: settingsv1alpha1.AddToScheme,
-		},
-	).Register(Registry, Scheme); err != nil {
-		panic(err)
-	}
-
-	Versions = append(Versions, storagev1.SchemeGroupVersion, storagev1beta1.SchemeGroupVersion)
-	if err := announced.NewGroupMetaFactory(
-		&announced.GroupMetaFactoryArgs{
-			GroupName:              storagev1.GroupName,
-			VersionPreferenceOrder: []string{storagev1.SchemeGroupVersion.Version, storagev1beta1.SchemeGroupVersion.Version},
-		},
-		announced.VersionToSchemeFunc{
-			storagev1.SchemeGroupVersion.Version:      storagev1.AddToScheme,
-			storagev1beta1.SchemeGroupVersion.Version: storagev1beta1.AddToScheme,
-		},
-	).Register(Registry, Scheme); err != nil {
-		panic(err)
-	}
+	Scheme.SetVersionPriorityOrDie(corev1.SchemeGroupVersion)
+	Scheme.SetVersionPriorityOrDie(admissionv1alpha1.SchemeGroupVersion)
```

Comment:
I was surprised these weren't done in `k8s.io/client-go/kubernetes/scheme#AddToScheme()`, at least as an initial ordering

Reply:
> I was surprised all of these weren't done in k8s.io/client-go/kubernetes/scheme#AddToScheme(), at least as an initial ordering

That one is generated, isn't it?  Also, the generated client knows what its expects, so preference there doesn't really mean anything.  The real question is why anyone would try to wire up to that client....

Reply:
hmm... is the setting of priority order here just to make `IsGroupRegistered` and `IsVersionRegistered` do reasonable things? (c.f. https://github.com/kubernetes/kubernetes/pull/63364/files#r186214718)